portfolio/src/main/webapp/js/main.js

```diff
@@ -23,16 +23,55 @@ $(document).ready(function(){
 })
 
 function getComments() {
-  let max = document.getElementById('num-comments').value;
-  fetch('/list-comments?max=' + max).then(response => response.json()).then((comments) => {
-    const commentsListElement = document.getElementById('comments-container');
-    commentsListElement.innerHTML = '';
-    comments.forEach(comment => {
-      commentsListElement.appendChild(createCommentElement(comment));
+  let numCommentsPerPage = document.getElementById('num-comments').value;
+  let numPages = setupPagination(numCommentsPerPage);
+  numPages.then(() => {
+    fetch('/list-comments?num-comments=' + numCommentsPerPage + '&page-num=' + currentPage())
+    .then(response => response.json())
+    .then((comments) => {
+      const commentsListElement = document.getElementById('comments-container');
+      commentsListElement.innerHTML = '';
+      comments.forEach(comment => {
+        commentsListElement.appendChild(createCommentElement(comment));
+      });
     });
   });
 }
 
+async function setupPagination(numCommentsPerPage) {
+  return fetch('/num-comments').then(response => response.text()).then((numCommentsString) => {
+    let numComments = parseInt(numCommentsString);
+    /* default is one page */
+    let numPages = Math.max(Math.ceil(numComments/numCommentsPerPage), 1);
+    const pageSelectElement = document.getElementById('page-num');
+    
+    if(pageSelectElement.options.length != numPages) {
+      updateOptions(pageSelectElement, numPages);
+    }
+  });
+}
+
+function updateOptions(pageSelectElement, numPages) {
+  let savedPage = pageSelectElement.options.length > 0 ? pageSelectElement.value : 1;
```

Comment:
Using Math.ceil might improve readability here. Same at line 68.

Reply:
I realized for line 55, I no longer need to check for this condition because the default length of options is 1. For line 68, I don't think Math.ceil makes sense. It's checking if the saved page is a valid page and if not taking you to the first page. However, I think it actually makes sense to take you to the last page if invalid since you were looking for a page > numPages so I will make that change.